reports/models.py

```diff
@@ -61,24 +61,44 @@ class BaseReportModel(models.Model):
     organization = models.ForeignKey(ORG_MODEL, on_delete=models.CASCADE)
 
     created_by = models.ForeignKey(
-        settings.AUTH_USER_MODEL, blank=True, null=True,
+        settings.AUTH_USER_MODEL,
+        blank=True,
+        null=True,
         on_delete=models.SET_NULL
     )
     created_at = models.DateTimeField(default=timezone.now, editable=False)
     config = JSONField(blank=True, default={})
-    emails = ArrayField(models.EmailField(max_length=255), blank=True,
-                        null=True)
+    emails = ArrayField(
+        models.EmailField(max_length=255), blank=True, null=True
+    )
     deleted = models.BooleanField(default=False)
 
     class Meta:
         abstract = True
 
 
 class Report(BaseReportModel):
+    STATUS_RUNNING = 'running'
+    STATUS_COMPLETED = 'completed'
+    STATUS_FAILED = 'failed'
+    STATUS_SKIPPED = 'skipped'
+
+    STATUS_CHOICES = (
+        (STATUS_RUNNING, 'Running'),
+        (STATUS_COMPLETED, 'Completed'),
+        (STATUS_FAILED, 'Failed'),
+        (STATUS_SKIPPED, 'Skipped'),
+    )
+
     start_datetime = models.DateTimeField()
     end_datetime = models.DateTimeField()
-    document = models.FileField(upload_to=report_upload_to, blank=True,
-                                null=True, max_length=1024)
+    document = models.FileField(
+        upload_to=report_upload_to, blank=True, null=True, max_length=1024
+    )
+    status = models.CharField(
+        max_length=256, choices=STATUS_CHOICES, default=STATUS_RUNNING
+    )
+    origin_report_id = models.PositiveIntegerField(null=True)
```

Comment:
Why not hav a foreign key instead?
And this name is quite confusing. Maybe just `schedule_id` ? @ross-urban your help is needed here 😄